Wrap long cell labels on Android

dcalhoun opened the pull request:
## Related PRs

- https://github.com/wordpress-mobile/gutenberg-mobile/pull/4475

## Description
Fixes #37858. Prior to this change, the cell label container was disallowed from shrinking to accommodate its siblings. This caused longer labels to force the sibling elements out of the viewport causing them to be invisible or cut-off.

Additional changes were applied to the link suggestion styles to ensure the sibling of cell icons do not encroach upon the now static width icon.

Lastly, https://github.com/WordPress/gutenberg/issues/37995 was created to capture an existing, unrelated issue identified while testing this work. 

## How has this been tested?
<details><summary>Case 1: Long Label Wraps</summary>

1. Change the device's language to one that generally requires more space than
   English, e.g. Spanish.
1. Open the block editor.
1. Add an Embed block.
1. Set a valid YouTube URL.
1. Open the Embed block settings.
1. ℹ️ Verify the _Resize for smaller devices_ toggle is visible.

</details>

<details><summary>Case 2: Scaled Text Controls Remain Accessible</summary> 

1. Set the device's OS text size to the largest available.
1. Open the block editor.
1. Add a Cover block.
1. Set an image for the Cover block.
1. Open the Cover block settings.
1. ℹ️ Verify the _Opacity_ slider displays and remains accessible. See https://github.com/WordPress/gutenberg/pull/20768#pullrequestreview-372353894 for additional details. 

</details>

<details><summary>Case 3: Link Picker Aligned Correctly</summary>

1. Copy a URL to the device clipboard.
1. Open the block editor.
1. Add a Paragraph block.
1. Add a link to the paragraph.
1. ℹ️ Verify the link suggestion is correctly aligned, i.e. the proper space is displayed between the left-side icon and right-side text, the link suggestion aligns with the search field above it.

</details>

<details><summary>Case 4: Android Media Options Aligned Correctly</summary>

1. Open the block editor on Android. 
1. Add an Image block. 
1. Tap the block to set media. 
1. ℹ️ Verify the listed options are correctly aligned, both the icons and label text. See https://github.com/WordPress/gutenberg/pull/20768 for additional details. 

</details>

## Screenshots <!-- if applicable -->

<details><summary>Long Label Wraps</summary>

| Android | iOS |
| - | - |
| ![Android cell label wraps](https://user-images.githubusercontent.com/438664/149593494-27a101cb-cb9f-4fda-a412-cc39c8bb9e76.jpg) | ![iOS cell label wraps](https://user-images.githubusercontent.com/438664/149593523-0cd81261-61dc-4612-b078-20c0ed77725d.PNG) |

</details>

<details><summary>Scaled Text Controls Remain Accessible</summary>

| Android | iOS |
| - | - |
| ![Android slider large text renders accessibly](https://user-images.githubusercontent.com/438664/149593598-f9b6e632-953e-46e0-a940-09ac9a0ffee5.jpg) | ![iOS slider text render accessibly](https://user-images.githubusercontent.com/438664/149593632-7086e32f-8697-4358-a00a-a00413698dbe.PNG) |

</details>

<details><summary>Link Picker Suggestion Aligned Correctly</summary>

| Before | After |
| - | - |
| ![Link suggestion misaligned before changes](https://user-images.githubusercontent.com/438664/149592954-4f6acff7-768d-4734-b52f-e0b986976932.jpg) | ![Link suggestion correctly aligned after changes](https://user-images.githubusercontent.com/438664/149592972-04b1f00e-027b-4e40-9538-10280a4ee340.jpg) |

</details>

<details><summary>Media Options Aligned Correctly</summary>

![Android media options are aligned correctly.](https://user-images.githubusercontent.com/438664/149594875-94742eb4-7e6e-4668-b16e-8a1959fe2daa.jpg)

</details>

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->
- [x] I've updated related schemas if appropriate. <!-- Reference: https://github.com/WordPress/gutenberg/tree/trunk/schemas -->

dcalhoun committed:
Wrap long cell labels on Android

Prior to this change, the cell label container was disallowed from
shrinking to accommodate its siblings. This caused longer labels to
force the sibling elements out of the viewport causing them to be
invisible or cut-off.

Additional changes were applied to the link suggestion styles to ensure
the sibling of cell icons do not encroach upon the now static width
icon.

dcalhoun committed:
Add change log entry

fluiddot commented on packages/components/src/mobile/bottom-sheet/styles.native.scss:
```diff
@@ -103,7 +103,13 @@
+.cellRowIcon {
 	flex-shrink: 0;
+	flex-basis: auto;
+	flex-grow: 0;
```
I'm wondering if we need to define these two style properties if both have their default value, and are not overriding already defined ones:
- `flex-basis` default value is `auto` ([reference](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-basis), couldn't find if the value for RN is the same as web but I understand that `auto` is the one that makes more sense)
- `flex-grow` default value is `0` ([reference](https://reactnative.dev/docs/layout-props#flexgrow))

What do you think?

dcalhoun replied:
Agreed. I removed the redundant style declarations in https://github.com/WordPress/gutenberg/pull/37993/commits/f0866dd92394410445b39dfad3dc4e2fe72233d6.

fluiddot reviewed (approved):
LGTM 🎊 !

I checked that all test cases pass and added some minor comments (none of them should be considered as blockers) in relation to the third case and within the file changes:
- **Case 1:** Long Label Wraps ✅ 
- **Case 2:** Scaled Text Controls Remain Accessible ✅ 
- **Case 3:** Link Picker Aligned Correctly ✅ _(with a comment)_ 
I think it makes sense to have the same padding on both sides, however, I'm wondering whether the content (i.e. icon and link as a single item) under the separator line is expected to be aligned with it (see attached screenshot), wdyt?

<img width="781" alt="Screenshot 2022-01-17 at 17 03 21" src="https://user-images.githubusercontent.com/14905380/149813261-7308276c-a2c7-4aca-b4f8-6547854bad9f.png">

- **Case 4: Android Media Options Aligned Correctly** ✅ 

_Tested on Samsung Galaxy S20 FE 5G (Android 10) and Simulator - iPhone 12 Pro Max (iOS 15.0)._

dcalhoun committed:
Remove unnecessary style declarations

These styles are the default values and redundant.

dcalhoun committed:
Expand link suggestion title to parent width

Force link suggestion title to fill parent width, which causes the link
suggestion to match the parent separator line width.

dcalhoun commented:
> * **Case 3:** Link Picker Aligned Correctly ✅ _(with a comment)_
>   I think it makes sense to have the same padding on both sides, however, I'm wondering whether the content (i.e. icon and link as a single item) under the separator line is expected to be aligned with it (see attached screenshot), wdyt?

Good question. It is difficult to tell from the original implementation images and discussion within https://github.com/wordpress-mobile/gutenberg-mobile/pull/2484. However, I went ahead and updated the styles to ensure the link suggestion title does align with the right side of the separator line. 

| Before | After |
| - | - |
| ![android before](https://user-images.githubusercontent.com/438664/149838878-d30fea52-903d-4e59-b91b-0f4027eff09b.jpg) | ![android after](https://user-images.githubusercontent.com/438664/149838890-36d44df3-a227-4bfd-9d66-6deed6a42fc2.jpg) |
| ![iOS before](https://user-images.githubusercontent.com/438664/149838910-a19c870f-10a9-499b-9116-05c31c20f60e.PNG) |  ![iOS after](https://user-images.githubusercontent.com/438664/149838919-26207a9a-ebed-4154-b478-f06c177fde85.PNG) |

I plan to move forward with merging this bug fix, but @iamthomasbishop please let me know if you disagree with the alignment change in the images.

iamthomasbishop commented:
Looks good to me, thanks @dcalhoun!
